moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -342,8 +394,74 @@ bool IKFastKinematicsPlugin::initialize(const std::string& robot_description, co
 {
   setValues(robot_description, group_name, base_name, tip_name, search_discretization);
 
-  ros::NodeHandle node_handle("~/" + group_name);
+  rdf_loader::RDFLoader rdf_loader(robot_description_);
+  const boost::shared_ptr<srdf::Model>& srdf = rdf_loader.getSRDF();
+  // const std::shared_ptr<urdf::ModelInterface>& urdf_model = rdf_loader.getURDF();
+  const boost::shared_ptr<urdf::ModelInterface>& urdf_model = rdf_loader.getURDF();
+
+  if (!urdf_model || !srdf)
+  {
+    ROS_ERROR_NAMED(name_, "URDF and SRDF must be loaded for KDL kinematics solver to work.");
+    return false;
+  }
+
+  robot_model::RobotModelPtr robot_model;
+  robot_model.reset(new robot_model::RobotModel(urdf_model, srdf));
+  robot_state_.reset(new robot_state::RobotState(robot_model));
+
+  // This IKFast solution was generated with IKFAST_TIP_FRAME_ and IKFAST_BASE_FRAME_.
+  // It is often the case that fixed joints are added to these links to model things like
+  // a robot mounted on a table or a robot with an end effector attached to the last link.
+  // To support these use cases, we store the transform from the IKFAST_BASE_FRAME_ to the
+  // base_frame_ and ikfast_tip_frame_to the tip_frame_ and transform to the input pose accordingly
+  tip_transform_required_ = !robot_state::Transforms::sameFrame(IKFAST_TIP_FRAME_, tip_frame_);
+  base_transform_required_ = !robot_state::Transforms::sameFrame(IKFAST_BASE_FRAME_, base_frame_);
+
+  // TODO(mlautman): Check for non-fixed joints between the ikfast tip and base frames and the group tip
+  //                 and base frames. If any are found, we should throw an error and return.
+  if (tip_frame_ != IKFAST_TIP_FRAME_)
```

Comment:
Why don't you check for additional joints here? The request should be rejected as soon as the frames are not rigidly connected to each other.